conversant/demo/streamlit_example.py

```diff
@@ -317,20 +327,12 @@ def update_prompt_from_json() -> None:
             """
             )
 
-            # Invokes partial reply until the final uterrance chunk is
-            # reached and redraws the corresponding text box each time
-            # a response is returned.
-            if "is_final_chunk" in st.session_state:
-                if (
-                    not st.session_state.is_final_chunk
-                    and st.session_state.rerun_count
-                    < st.session_state.bot.get_max_reruns()
-                ):
-                    st.session_state.rerun_count += 1
-                    response, is_final_chunk = st.session_state.bot.partial_reply(
-                        query="", is_from_scratch=False
-                    )
-                    st.session_state.is_final_chunk = is_final_chunk
-                    if is_final_chunk:
-                        st.session_state.text_input_disabled = False
+            if "partial_reply_generator" in st.session_state:
+                st.session_state.text_input_disabled = True
+                partial_chunk = peek(st.session_state.partial_reply_generator)
+                if partial_chunk is not None:
+                    st.experimental_rerun()
```

Comment:
This is a neat way of using the partial reply and having it work for Streamlit! 🎉